src/registrar/management/commands/update_first_ready.py

```diff
@@ -0,0 +1,28 @@
+import logging
+from django.core.management import BaseCommand
+from registrar.management.commands.utility.terminal_helper import PopulateScriptTemplate, TerminalColors
+from registrar.models import Domain, TransitionDomain
+
+logger = logging.getLogger(__name__)
+
+class Command(BaseCommand, PopulateScriptTemplate):
+    help = "Loops through each domain object and populates the last_status_update and first_submitted_date"
+
+    def handle(self, **kwargs):
+        """Loops through each valid Domain object and updates it's first_ready value if it is out of sync"""
+        filter_conditions={"state__in":[Domain.State.READY, Domain.State.ON_HOLD, Domain.State.DELETED]}
+        self.mass_update_records(Domain, filter_conditions, ["first_ready"], verbose=True, custom_filter=self.should_update)
```

Comment:
While I think that would be nice for anyone that understands querysets well, it might create a bit of confusion for anyone that doesn't really get querysets (which I personally find confusing). What I like about this helper is that it shields the user from a lot of the more in-depth queryset stuff. If anything I would want to go the opposite direction and refactor `mass_update_records` to just use a  `filter_records` function in place of `filter_conditions` that allows the user to express their filter conditions in regular python, rather than having to deal with django field lookups. A lot like how you pulled out the update logic into `update_record`.

Reply:
Main downside I see for that is that idk how it would impact performance. I imagine Django's ORM is doing some kind of optimization in the background that we might be losing out on.